### PR TITLE
Added API to retrieve network statistics.

### PR DESCRIPTION
**Pull request by mykhailo-kuchma**
This API should enable users to query network statistics and aggregate
the statistics by buckets (Wifi/Cellular/etc.) when needed.

Resolves: OLPEDGE-1761

Signed-off-by: Mykhailo Kuchma <ext-mykhailo.kuchma@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/include/olp/core/http/Network.h [hunk 1]
```diff
@@ -90,7 +105,24 @@ class CORE_API Network {
...
    */
-  virtual void SetDefaultHeaders(Headers) {}
+  virtual void SetDefaultHeaders(Headers headers);
+
+  /**
+   * @brief Set the current statistics bucket.
+   *
+   * @param[in] bucked_id The bucket ID.
+   */
+  virtual bool SetCurrentBucket(uint8_t bucket_id);
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/http/Network.h:115]: Do we need to return anything? Can this operation fail?
If yes, then please add doxygen explaining when do we return true and when do we return false.

File: olp-cpp-sdk-core/include/olp/core/http/Network.h [hunk 2]
```diff
@@ -90,7 +105,24 @@ class CORE_API Network {
...
+
+  /**
+   * @brief Get the statistics for bucket.
+   *
+   * By default returns statistics for a default bucket.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/http/Network.h:120]:
```suggestion
   * By default, this returns the statistics for the default bucket, with the ID 0.
```

File: olp-cpp-sdk-core/include/olp/core/http/Network.h [hunk 3]
```diff
@@ -90,7 +105,24 @@ class CORE_API Network {
...
+
+  /**
+   * @brief Set the current statistics bucket.
+   *
+   * @param[in] bucked_id The bucket ID.
+   */
+  virtual bool SetCurrentBucket(uint8_t bucket_id);
+
+  /**
+   * @brief Get the statistics for bucket.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/http/Network.h:118]:
```suggestion
   * @brief Get the statistics for a bucket.
```

File: olp-cpp-sdk-core/include/olp/core/http/Network.h [hunk 4]
```diff
@@ -90,7 +105,24 @@ class CORE_API Network {
...
+   *
+   * By default returns statistics for a default bucket.
+   *
+   * @param[in] bucked_id The bucket ID.
+   * @return Statistics which contains the statistic for a requested bucket.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/http/Network.h:123]:
```suggestion

   * @return Statistics which contains the statistic for the requested bucket.
```
NOTE: Extra newline between parameters and return.

File: olp-cpp-sdk-core/src/http/Network.cpp [hunk 5]
```diff
@@ -52,6 +52,14 @@ std::shared_ptr<Network> CreateDefaultNetworkImpl(size_t max_requests_count) {
 }
 }  // namespace
 
+void Network::SetDefaultHeaders(Headers /*headers*/) {}
```
Comment (andescu) [inline olp-cpp-sdk-core/src/http/Network.cpp:55]:
```suggestion
void Network::SetDefaultHeaders(Headers) {}
```

File: olp-cpp-sdk-core/src/http/Network.cpp [hunk 6]
```diff
@@ -52,6 +52,14 @@ std::shared_ptr<Network> CreateDefaultNetworkImpl(size_t max_requests_count) {
...
+
+bool Network::SetCurrentBucket(uint8_t /*bucket_id*/) { return false; }
```
Comment (andescu) [inline olp-cpp-sdk-core/src/http/Network.cpp:57]:
```suggestion
bool Network::SetCurrentBucket(uint8_t) { return false; }
```

File: olp-cpp-sdk-core/src/http/Network.cpp [hunk 7]
```diff
@@ -52,6 +52,14 @@ std::shared_ptr<Network> CreateDefaultNetworkImpl(size_t max_requests_count) {
...
+
+Network::Statistics Network::GetStatistics(uint8_t /*bucket_id*/) {
```
Comment (andescu) [inline olp-cpp-sdk-core/src/http/Network.cpp:59]:
```suggestion
Network::Statistics Network::GetStatistics(uint8_t) {
```

File: olp-cpp-sdk-core/include/olp/core/http/Network.h [hunk 8]
```diff
@@ -51,6 +51,21 @@ class CORE_API Network {
...
+   */
+  struct Statistics {
+    /// The total bytes downloaded including size of headers and payload.
+    uint64_t bytes_downloaded;
+    /// The total bytes uploaded including size of headers and payload.
+    uint64_t bytes_uploaded;
+    /// The total number of requests made by network.
+    uint32_t total_requests;
+    /// The total number of requests failed.
+    uint32_t total_failed;
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/http/Network.h:66]:
```suggestion
    /// The total bytes downloaded including size of headers and payload.
    uint64_t bytes_downloaded{0u};
    /// The total bytes uploaded including size of headers and payload.
    uint64_t bytes_uploaded{0u};
    /// The total number of requests made by network.
    uint32_t total_requests{0u};
    /// The total number of requests failed.
    uint32_t total_failed{0u};
```

File: olp-cpp-sdk-core/src/http/Network.cpp (hunk 7 above)
Comment (mykhailo-kuchma) [reply olp-cpp-sdk-core/src/http/Network.cpp:59]: a variable name is placed for consistency

File: olp-cpp-sdk-core/src/http/Network.cpp (hunk 6 above)
Comment (mykhailo-kuchma) [reply olp-cpp-sdk-core/src/http/Network.cpp]: a variable name is placed for consistency

File: olp-cpp-sdk-core/src/http/Network.cpp (hunk 5 above)
Comment (mykhailo-kuchma) [reply olp-cpp-sdk-core/src/http/Network.cpp:55]: a variable name is placed for consistency

**Pull request merged**